examples/stable_diffusion_v2/controlnet_image2image.py

```diff
@@ -187,6 +187,24 @@ def decode_and_save_result(args, samples, detected_map, outpath, filename):
 
     logger.info(f"Save result to {outpath} done.")
 
+    # second sampling, no need upload this part
+    logging.info("Start the second sampling:")
+    samples, intermediates = sampler.sample(
+        ddim_steps,
+        num_samples,
+        shape,
+        cond,
+        verbose=False,
+        eta=eta,
+        unconditional_guidance_scale=scale,
+        unconditional_conditioning=un_cond,
+    )
+    logging.info("The second sampling, done")
+
+    decode_and_save_result(args, samples, detected_map, outpath, "results")
```

Comment:
`decode_and_save_result` is executed twice: here and at line 186. Also, `decode_and_save_result` doesn't need to be a nested function. Move it outside `main`.

Reply:
Oh this part is for testing the inferencing time, and there is no need to upload this part.

Reply:
Will move this function outside of the `main`.

Reply:
Updated.